ReactAndroid/src/main/java/com/facebook/react/views/image/events/ImageLoadedEvent.java

```diff
@@ -0,0 +1,40 @@
+/**
+ * Copyright (c) 2015-present, Facebook, Inc.
+ * All rights reserved.
+ *
+ * This source code is licensed under the BSD-style license found in the
+ * LICENSE file in the root directory of this source tree. An additional grant
+ * of patent rights can be found in the PATENTS file in the same directory.
+ */
+
+package com.facebook.react.views.image.events;
+
+import com.facebook.react.bridge.Arguments;
+import com.facebook.react.uimanager.events.Event;
+import com.facebook.react.uimanager.events.RCTEventEmitter;
+
+public class ImageLoadedEvent extends Event<ImageLoadedEvent> {
+
+  public static final String EVENT_NAME = "topImageLoad";
```

Comment:
So as @mkonicek pointed out, iOS actually has five different image events. I'd looked into this earlier. Fresco can easily do onLoad, onLoadEnd, and onLoadStart. It can do onError some of the time. It can't yet do onProgress.

You should be able to pile all those types here. Don't use a Java enum, they're not efficient. Instead do this:

```java
​  @IntDef({ON_ERROR, ON_LOAD, ON_LOAD_END, ON_LOAD_START, ON_PROGRESS})
  @Retention(RetentionPolicy.SOURCE)
  @interface ImageEventType {}

  static final int ON_ERROR = 1;
  static final int ON_LOAD = 2;
  static final int ON_LOAD_END = 3;
  static final int ON_LOAD_START = 4;
  static final int ON_PROGRESS = 5;

  private final int mEventType;
```